plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterAttribute.php

```diff
@@ -308,7 +306,7 @@ private function get_attribute_counts( $block, $slug, $query_type ) {
 			$query_vars['tax_query'] = ProductCollectionUtils::remove_query_array( $query_vars['tax_query'], 'taxonomy', $slug );
 		}
 
-		$counts           = $filters->get_attribute_counts( $query_vars, $slug );
+		$counts           = wc_get_container()->get( FilterDataProvider::class )->get_attribute_counts( $query_vars, $slug );
```

Comment:
How is the instance (of instances) of `ProductFilterAttribute` obtained? If it's with `new` or another mechanism outside the dependency injection container, then it's fine to use `wc_get_container` to grab the dependencies. But it `ProductFilterAttribute` itself is registered in the container, then dependencies should be passed in the `init` method and cached locally (as `FilterDataProvider` does with `FilterClausesGenerator`, for example). And the same with the other classes that use `wc_get_container`.

Reply:
@Konamiman `ProductFilterAttribute` is managed Blocks: https://github.com/woocommerce/woocommerce/blob/3f0cf66ba050a2258e79a03590bf7eaba9343ae5/plugins/woocommerce/src/Blocks/BlockTypesController.php#L100-L108

Reply:
I see, so `ProductFilterAttribute` is not handled by the container. In that case it's fine to use `wc_get_container` to retrieve its dependencies.